ios/testmanagerd/proxydispatcher.go

```diff
@@ -17,6 +18,13 @@ type proxyDispatcher struct {
 }
 
 func (p proxyDispatcher) Dispatch(m dtx.Message) {
+	defer func() {
+		if r := recover(); r != nil {
+			p.testListener.err = fmt.Errorf("Dispatch: %s", r)
+			p.testListener.TestRunnerKilled()
```

Comment:
We don't know whether the TestRunner has been killed, only a message dispatch failed. Unless part of the recovery process is to kill the TestRunner.

Reply:
Yes, this should definitely not introduce the side effect of killing the runner.

And are you trying to recover from a potential panic in the `extract...` functions of this file? If that's the case please use safe [Type Assertions](https://go.dev/ref/spec#Type_assertions) (that should be the default anyways I guess)

Reply:
> We don't know whether the TestRunner has been killed

That's why we assume it's killed or should be killed by us. There is no partial recovery that can save a test run from a panic.

Reply:
> Yes, this should definitely not introduce the side effect of killing the runner.

Removed it.

> And are you trying to recover from a potential panic in the extract... functions of this file? 

Yes but not due to type assertions. The panic recovered in `Unarchive()` can still cause nil derefence errors after the fact.